code/modules/security_levels/keycard_authentication.dm

```diff
@@ -185,6 +195,25 @@
 						ERT_Announce(ert_reason , event_triggered_by, 1)
 			else
 				trigger_armed_response_team(new /datum/response_team/amber) // No admins? No problem. Automatically send a code amber ERT.
+		if("Gamma Alert")
+			atom_say("Gamma Alert request transmitted!")
+			GLOB.command_announcer.autosay("Gamma Alert request transmitted. Reason: [gamma_reason]", name)
+			print_centcom_report(gamma_reason, station_time_timestamp() + " Gamma Alert Request")
+
+			var/fullmin_count = 0
+			for(var/client/C in GLOB.admins)
+				if(check_rights(R_EVENT, 0, C.mob))
+					fullmin_count++
+			if(fullmin_count)
+				GLOB.gamma_request_answered = TRUE
+				GAMMA_Announce(gamma_reason , event_triggered_by, 0)
+				gamma_reason = null
+				SSblackbox.record_feedback("nested tally", "keycard_auths", 1, list("gamma", "called"))
+				spawn(3000)
```

Comment:
This might be better as a callback or invoke async. We try to avoid using spawns.
Yes, some old code, including the code you are using, still uses them.

Reply:
Definitely should be an addtimer check here

Reply:
removed it entirely, a 5-minute reminder for this was pointless.